dev/dev_setup.sh

```diff
@@ -88,29 +117,47 @@ function logged {
     fi
 }
 
-function update_submodules {
+update_submodules() {
     cd $ROOT
+    if ! is_installed git; then
+        notice "Installing git"
+        case $CRYPTOL_PLATFORM in
+            $UBUNTU20 | $UBUNTU22) logged apt-get install -y git;;
+            $MACOS12 | $MACOS14) logged brew install git && USED_BREW=true;;
+            *) unreachable "Unsupported platform; did not install git"; return;;
+        esac
+    fi
     notice "Updating submodules"
     logged git submodule update --init
 }
 
-function install_ghcup {
+install_ghcup() {
     if ! is_installed ghcup; then
-        notice "Installing GHCup via Homebrew"
-        # Manually install Haskel toolchain. If this doesn't work, try
+        # Manually install Haskell toolchain. If this doesn't work, try
         # using the install script at https://www.haskell.org/ghcup/
-        if [ $CRYPTOL_PLATFORM = $MACOS12 ] || [ $CRYPTOL_PLATFORM = $MACOS14 ]; then
-            logged brew install ghcup
-            USED_BREW=true
-        else
-            notice "Did not install GHCup. This script only supports macOS 14"
-        fi
+        case $CRYPTOL_PLATFORM in
+            $MACOS12 | $MACOS14)
+                notice "Installing GHCup via Homebrew"
+                logged brew install ghcup && USED_BREW=true;;
+            $UBUNTU20 | $UBUNTU22)
+                notice "Installing any missing GHCup dependencies via apt-get"
+                logged apt-get install -y build-essential curl libffi-dev libffi7 libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5
+
+                notice "Installing GHCup via direct download"
+                logged curl --proto '=https' --tlsv1.2 -sSfL -o ghcup "$GHCUP_URL/$(uname -m)-linux-ghcup"
```

Comment:
note: I couldn't find a version of GHCup for linux that came via a normal package manager, hence the manual install of dependencies + downloading and running a binary straight from the internet 😬

Reply:
This has been sadly the standard that we have done for most of our docker container/scripts thus far. I also couldn't find a better source.

Reply:
To be fair, `ghcup` explicitly bills itself as an alternative to your system package manager that is specific to GHC. Both `ghcup` and your system package manager will, in some sense, download a binary straight from the Internet—it's just a matter of whether you trust the tool that's providing it.